src/cleanlab_tlm/utils/rag.py

```diff
@@ -892,6 +905,189 @@ def __init__(
         self.context_identifier = context_identifier
         self.response_identifier = response_identifier
 
+        # Compile and validate the eval
+        self.mode = self._compile_mode(mode, criteria, name)
+
+    def _compile_mode(self, mode: Optional[str], criteria: str, name: str) -> str:
```

Comment:
This _compile_mode method needs to be written more carefully to avoid unintentionally breaking all our tests. A lot of the userwarnings will be thrown as errors during automated testing.